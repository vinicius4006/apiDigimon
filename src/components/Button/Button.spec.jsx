import { render, screen } from '@testing-library/react';
import { click } from '@testing-library/user-event/dist/click';
import { Button } from '.';

describe('<Button />', () => {
    it('should render the button with the text "LOAD MORE +" ', () => {
        render(<Button text="Load More +"/>);
        expect.assertions(1);


        const button = screen.getByRole('button', { name: /load more +/i });
        expect(button).toHaveAttribute('class', 'btn btn-transition btn-styles');
    });

    it('should call function on button click', () => {

        const fn = jest.fn();
        render(<Button text="Load More +" loadMoreDigi={fn} />);

        const button = screen.getByRole('button', { name: /load more +/i });

        click(button);
        //fireEvent.click(button);
        expect(fn).toHaveBeenCalledTimes(1);
    });

    it('should be disabled when disabled is true', () => {

        render(<Button text="Load More +" notMoreDigi={true} />);

        const button = screen.getByRole('button', { name: /load more +/i });

        expect(button).toBeDisabled();
    });

    it('should be enabled when disabled is false', () => {

        render(<Button text="Load More +" notMoreDigi={false} />);

        const button = screen.getByRole('button', { name: /load more +/i });

        expect(button).toBeEnabled();
    });
});