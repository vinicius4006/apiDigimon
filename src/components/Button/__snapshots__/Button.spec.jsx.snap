// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot 1`] = `
<button
  class="btn btn-transition btn-styles"
>
  Load More +
</button>
`;
